gpt4all-chat/database.cpp

```diff
@@ -890,11 +890,11 @@ void Database::scanDocuments(int folder_id, const QString &folder_path)
     qDebug() << "scanning folder for documents" << folder_path;
 #endif
 
-    static const QList<QString> extensions { "txt", "doc", "docx", "pdf", "rtf", "odt", "html", "htm",
+    static const QList<QString> extensions { "txt", "doc", "pdf", "rtf", "odt", "html", "htm",
     "xls", "xlsx", "csv", "ods", "ppt", "pptx", "odp", "xml", "json", "log", "md", "org", "tex", "asc", "wks",
     "wpd", "wps", "wri", "xhtml", "xht", "xslt", "yaml", "yml", "dtd", "sgml", "tsv", "strings", "resx",
     "plist", "properties", "ini", "config", "bat", "sh", "ps1", "cmd", "awk", "sed", "vbs", "ics", "mht",
-    "mhtml", "epub", "djvu", "azw", "azw3", "mobi", "fb2", "prc", "lit", "lrf", "tcr", "pdb", "oxps",
+    "mhtml", "djvu", "azw", "azw3", "fb2", "prc", "lit", "lrf", "tcr", "pdb", "oxps",
     "xps", "pages", "numbers", "key", "keynote", "abw", "zabw", "123", "wk1", "wk3", "wk4", "wk5", "wq1",
     "wq2", "xlw", "xlr", "dif", "slk", "sylk", "wb1", "wb2", "wb3", "qpw", "wdb", "wks", "wku", "wr1",
     "wrk", "xlk", "xlt", "xltm", "xltx", "xlsm", "xla", "xlam", "xll", "xld", "xlv", "xlw", "xlc", "xlm",
```

Comment:
Wow... why do we try to support so many extensions? People have tried Excel spreadsheets with limited success (since they usually don't contain plain English). Similar situation with CSV and TSV. PLIST, PDB, SLK, SYLK, WB1/2/3... these do not contain (meaningful) text. WPS is for Microsoft Works, which I highly doubt anyone is using in 2024...

Reply:
we can certainly limit it down. i thought it was good to go for as many types of text as possible thinking people would know what was in their files and leave it up to them to determine, but i was wrong. i admit it.

Reply:
feel free to remove...

Reply:
I propose we only list `.txt`, `.md`, and `.pdf`. More extensions can be added if requested - with code to actually parse them (e.g. I know you can convert many documents to txt with ebook-convert from Calibre, there must be a library that can do this too).

Reply:
Pandoc can do lots of conversions and offers a Haskell library (whatever that means) and a CLI.